pkg/boot/netboot/simple/simple.go

```diff
@@ -0,0 +1,43 @@
+// Copyright 2017-2018 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package simple
+
+import (
+	"context"
+	"fmt"
+	"io"
+	l "log"
+	"math"
+	"net/url"
+
+	"github.com/u-root/u-root/pkg/boot"
+	"github.com/u-root/u-root/pkg/boot/fit"
+	"github.com/u-root/u-root/pkg/curl"
+)
+
+// FetchAndProbe fetches the file at the specified URL and checks if it is an
+// Image file type rather than a config such as ipxe.
+// TODO: detect nonFIT multiboot and bzImage Linux kernel files
+func FetchAndProbe(ctx context.Context, u *url.URL, s curl.Schemes, verbose bool) ([]boot.OSImage, error) {
+	file, err := s.Fetch(ctx, u)
+	if err != nil {
+		return nil, err
+	}
+	var images []boot.OSImage
+
+	fimgs, err := fit.ParseConfig(io.NewSectionReader(file, 0, math.MaxInt64), verbose)
+	if err == nil {
+		for i := range fimgs {
+			images = append(images, &fimgs[i])
+		}
+	} else {
```

Comment:
did you want to keep going if this parseconfig fails?

Reply:
Yeah, similarly to netboot.go I figured more Image formats could populate the file.